projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-results/search-ai-results.component.ts

```diff
@@ -168,17 +168,24 @@ export class SearchAiResultsComponent extends PageComponent implements OnInit, O
 
   performAiSearch(): void {
     this._loading = true;
+
     this.searchAiService
       .ask({
         question: this.searchQuery,
-        nodeIds: this._selectedNodesState?.nodes?.length ? this._selectedNodesState.nodes.map((node) => node.entry.id) : []
+        nodeIds: this._selectedNodesState?.nodes?.length ? this._selectedNodesState.nodes.map((node) => node.entry.id) : [],
+        agentId: this._agentId
       })
       .pipe(
         switchMap((response) => this.searchAiService.getAnswer(response.questionId)),
+        retryWhen((errors) => this.aiSearchRetryWhen(errors)),
```

Comment:
I see errors in retryWhen has any type inside Observable, can we use some different type? Maybe there is some in built type to use? 
<img width="926" alt="image" src="https://github.com/user-attachments/assets/6319b0c2-7d42-4ef1-909a-883232e41a55">

Reply:
The same in 188th line